packages/salesforcedx-vscode-core/src/util/orgUtil.ts

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (c) 2022, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import { Aliases } from '@salesforce/core';
+import { channelService } from '../channels';
+import { nls } from '../messages';
+import { notificationService } from '../notifications';
+import { FileInfo, OrgList } from '../orgPicker';
+
+export async function setUpOrgExpirationWatcher() {
+  // Run once to start off with.
+  await checkForExpiredOrgs();
+
+  /*
+  Comment this out for now.  For now, we are only going to check once on activation,
+  however it would be helpful if we also checked once a day.  If we decide to also
+  check once a day, uncomment the following code.
+
+  // And run again once every 24 hours.
+  setInterval(async () => {
+    await checkForExpiredOrgs();
+  }, 1000 * 60 * 60 * 24);
+  */
+}
+
+export async function checkForExpiredOrgs() {
+  try {
+    const daysBeforeExpire = 5;
+    const today = new Date();
+    const daysUntilExpiration = new Date();
+    daysUntilExpiration.setDate(daysUntilExpiration.getDate() + daysBeforeExpire);
+
+    const orgList = new OrgList();
+    const authInfoObjects = await orgList.getAuthInfoObjects();
```

Comment:
In the [draft PR](https://github.com/forcedotcom/salesforcedx-vscode/pull/4297/files#diff-2b72c15ab3c2e3d66f269772e80aed9902d2d85d9acb2a462116d03541484d4fR51) that updates orgList to not read from the .sfdx directory, the return type of this method changes to return the OrgAuthorization type from core.  

FileInfo is a type that is created after reading the files in the ~/.sfdx directory.  

That change would require some reworking of this methods internals, to work with OrgAuthorization type and the AuthFields type for auth fields that are stored there in core v3, like expiration date: https://github.com/forcedotcom/salesforcedx-vscode/pull/4297/files#diff-2b72c15ab3c2e3d66f269772e80aed9902d2d85d9acb2a462116d03541484d4fR97

Reply:
Maybe this file could import and use AuthInfo.listAllAuthorizations()

Reply:
I've spoke with Ken about this, and the plan is to merge in now and not be held up, then pull develop into his branch and then address this issue.